lib/routines.go

```diff
@@ -0,0 +1,24 @@
+package lib
+
+import "sync"
+
+func ParallelApply[T comparable, K comparable](items []T, f func(item T) (*[]K, error)) *[]K {
+	var wg sync.WaitGroup
+	results := []K{}
+	for _, item := range items {
+		wg.Add(1)
+		go func(item T) {
+			defer wg.Done()
+			result, err := f(item)
+			if err != nil {
+				return
+			}
+			if result != nil {
+				results = append(results, *result...)
+			}
+		}(item)
+	}
+	wg.Wait()
+
+	return &results
+}
```

Comment:
I am using a generic method to run a function on each item in a slice, in parallel, and collect the results.